conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt

```diff
@@ -0,0 +1,1330 @@
+             type of the result is int:UnsignedN where N is the smallest such K via invalid assignment with inference.
+Labels: binary-bitwise-expr, DecimalNumber, int, int:Unsigned8, int:Unsigned16, int:Unsigned32, int:Signed8, 
+        int:Signed16, var
+
+function init() {
+    int a = 4343;
+    int b = 8;
+    var result1 = a & b;
+    int:Unsigned32 _ = result1; // @error the static type of the result is int
+
+    int d = 9876;
+    int:Unsigned32 e = 5454;
+    var result2 = d & e;
+    int:Unsigned16 _ = result2; // @error smallest K of the operands is 32
+
+    int g = 9876;
+    int:Unsigned16 h = 5454;
+    var result3 = g & h;
+    int:Unsigned8 _ = result3; // @error smallest K of the operands is 16
+
+    byte _ = result3; // @error smallest K of the operands is 16
+
+    int:Signed8 k = 12;
+    int:Signed16 m = 4434;
+    var result4 = k & m;
+    int:Signed8 _ = result4; // @error the static type of the result is int
+}
+
+Test-Case: output
+Description: Test nil lifting for bitwise and operator.
+Labels: binary-bitwise-expr, byte, DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8, 
+        int:Unsigned16, int:Unsigned32, numeric-nil-lifting, optional-type, unary-minus
+
+function init() {
+    int? a = 43;
+
+    byte? b = 5;
+
+    int:Signed32? c = -21474836;
+    int:Signed16? d = -454;
+    int:Signed8? e = 100;
+
+    int:Unsigned32? f = 21474836;
+    int:Unsigned16? g = 54522;
+    int:Unsigned8? h = 255;
+
+    printResult(a & b); // @output 1
+    printResult(a & c); // @output 40
+    printResult(a & d); // @output 42
+    printResult(a & e); // @output 32
+    printResult(a & f); // @output 0
+    printResult(a & g); // @output 42
+    printResult(a & h); // @output 43
+
+    printResult(b & b); // @output 5
+    printResult(b & c); // @output 4
+    printResult(b & d); // @output 0
+    printResult(b & e); // @output 4
+    printResult(b & f); // @output 4
+    printResult(b & g); // @output 0
+    printResult(b & h); // @output 5
+
+    printResult(c & c); // @output -21474836
+    printResult(c & d); // @output -21475288
+    printResult(c & e); // @output 100
+    printResult(c & f); // @output 4
+    printResult(c & g); // @output 20712
+    printResult(c & h); // @output 236
+
+    printResult(d & d); // @output -454
+    printResult(d & e); // @output 32
+    printResult(d & f); // @output 21474832
+    printResult(d & g); // @output 54330
+    printResult(d & h); // @output 58
+
+    printResult(e & e); // @output 100
+    printResult(e & f); // @output 4
+    printResult(e & g); // @output 96
+    printResult(e & h); // @output 100
+
+    printResult(f & f); // @output 21474836
+    printResult(f & g); // @output 33808
+    printResult(f & h); // @output 20
+
+    printResult(g & g); // @output 54522
+    printResult(g & h); // @output 250
+
+    printResult(h & h); // @output 255
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
```

Comment:
Don't we have error test for the static type with numeric nil lifting?

Reply:
Yes, we have. https://github.com/ballerina-platform/ballerina-spec/pull/941/files#diff-b61d4efe62f7b5e3b7b97e8277d5f19fa183e9101e70b39fb6e13b88c68c1e5dR716

Reply:
L#716